server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -2224,21 +2225,11 @@ public void deletePools(Collection<Pool> pools, Collection<String> alreadyDelete
                     this.enforcer.postUnbind(entitlement.getConsumer(), this, entitlement);
                 }
 
-                /* Save the consumer IDs and then detach all the keys in the consumerStackEnts map.
-                 * Otherwise during the status calculations, the facts proxy objects objects will be resolved
-                 * and the memory use will grow linearly with the number of consumers instead of remaining
-                 * constant as we calculate the status of each consumer.  See BZ 1584259 */
-                List<String> consumers = new ArrayList<String>(consumerStackedEnts.keySet().size());
-                for (Consumer c : consumerStackedEnts.keySet()) {
-                    consumers.add(c.getId());
-                }
-                this.consumerCurator.batchDetach(consumerStackedEnts.keySet());
-
-                log.info("Recomputing status for {} consumers", consumers.size());
+                log.info("Recomputing status for {} consumers", consumerStackedEnts.keySet().size());
 
                 // Recalculate status for affected consumers
-                for (List<String> subList : Lists.partition(consumers, 1000)) {
-                    for (Consumer consumer : this.consumerCurator.getConsumers(subList).list()) {
+                for (List<Consumer> subList : Iterables.partition(consumerStackedEnts.keySet(), 1000)) {
+                    for (Consumer consumer : subList) {
                         this.complianceRules.getStatus(consumer);
                         this.consumerCurator.detach(consumer);
```

Comment:
I would like to keep the comment starting on line 2227 about the purpose of the detach above this line.

Reply:
Done. I've reworded it a bit to make sense in the new context and moved it to the detach line as requested here.